test/src/400-ducc-ingest_images/main

```diff
@@ -1,54 +1,62 @@
-#!/bin/bash
+cvmfs_test_name="DUCC container ingestion"
+cvmfs_test_autofs_on_startup=false
+cvmfs_test_suites="ducc"
 
 registry=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 8 | head -n 1)
-recipe="$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 8 | head -n 1).yaml"
+recipe="$(pwd)/$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 8 | head -n 1).yaml"
 repository="$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 8 | head -n 1).cern.ch"
+ducc_log="$(pwd)/$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 8 | head -n 1).log"
+ducc_err="$(pwd)/$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 8 | head -n 1).err"
 
-clean_up() {
+ducc_test_400_clean_up() {
     printf "Cleaning up..."
-    
+
     mv /etc/cvmfs/default.local.bk /etc/cvmfs/default.local
     rm -f /etc/cvmfs/config.d/$repository.conf
     #delete the docker images
-    
+
     # delete the CVMFS repo
     printf "Deleting CVMFS repo..."
     cvmfs_server rmfs -f $repository >> /dev/null
     printf "done\n"
-    
-    export DOCKER2CVMFS_DOCKER_REGISTRY_PASS=""
-    
+
+    export DUCC_DOCKER_REGISTRY_PASS=""
+
     # delete the recipe file
     rm $recipe
-    
+    # delete the logs and error from ducc
+    rm $ducc_log
+    rm $ducc_err
+
     # stop and delete the registry container
     printf "Stopping and deleting docker registry..."
     docker stop $registry >> /dev/null
     docker rm $registry >> /dev/null
     printf "done\n"
-    
+
     printf "Reseting initial docker configuration..."
     rm -f /etc/docker/daemon.json
     mv /etc/docker/daemon.json.bk /etc/docker/daemon.json
-    
+
     systemctl restart docker
-    
+
     printf "done\n"
 }
 
+cvmfs_run_test() {
+    trap ducc_test_400_clean_up EXIT
 
-main() {
-    trap clean_up EXIT
-    
     # change the docker configuration to run thin images
-    printf "Changing the docker configuration to run thin images..."
-    
+    printf "*** Changing the docker configuration to run thin images..."
+
     # saving old docker configuration
+    mkdir -p /etc/docker
+    touch /etc/docker/daemon.json
     mv /etc/docker/daemon.json /etc/docker/daemon.json.bk
-    
+
     # installing thin image plugin
     docker plugin install --grant-all-permissions cvmfs/graphdriver
```

Comment:
This can fail and should then back out with an error code, i.e.

    docker plugin install ... | return 10

Reply:
This command usually fails, because the plugin is already installed

Reply:
In this case, the test should test whether the plugin is installed and only otherwise install it